### PR TITLE
Profile/reduxflow

### PR DESCRIPTION
**Pull request by josegocampo**
Fixed Redux's flow for the Profile Data.

Both userIdReducer and PostProfileReducer (which inclused getting the profille data from the API and also updating that data in the API) are done with their respective actions.  

So now you should be able to Login, see the profiles, they will appear corresponding to which user type is logged in (Groomer = 1, Client = 2), click the profile edit button, edit that data, save and go back to the profile with the updated information, which also should have updated the API.

That's about it. 

Let me know what bugs or performance optimization you guys find :)

### CODE REVIEW

File: src/components/pages/Home/RenderHomePage.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ function RenderHomePage() {
   return (
     <div className="flux" style={{ textAlign: 'center' }}>
       <div className="user-info">
-        <h1>Hi {userData?.name}</h1>
+        <h1>Hi {userData.name}</h1>
```
Comment (hmerritt) [inline src/components/pages/Home/RenderHomePage.js]: This change seems to be failing the tests.

I actually did the exact same thing myself thinking the `?` was a syntax error. It seems like the `?` is a sort of fallback which won't throw an error if the parameter `.name` does not exist.

**Review by hmerritt — CHANGES_REQUESTED** [review]
The test is failing - I figured out why though so you should be able to fix it quickly.

On the page `RenderHomePage.js` there should be a `?` after `userData` like this `userData?.name` - it acts as a fallback if `.name` does not exist.

File: src/components/pages/Home/RenderHomePage.js (hunk 1 above)
Comment (josegocampo) [reply src/components/pages/Home/RenderHomePage.js]: Thanks Harry!

Good catch. Fixing it and pushing.

**Review by hmerritt — APPROVED** [review]
Nice work expanding redux and rendering a different profile page based on user type.

File: src/api/index.js [hunk 2]
```diff
@@ -43,7 +43,6 @@ const getProfileData = authState => {
     return apiAuthGet(getAuthHeader(authState)).then(response => response.data);
   } catch (error) {
     return new Promise(() => {
-      console.log(error);
```
Comment (dvwhite) [inline src/api/index.js:46]: Thanks for removing the console log!

File: src/components/pages/ProfilePage/GroomerProfilePage.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ const GroomerProfilePage = ({ user }) => {
           <div className="groomers-mini">
             <div className="userinfo">
               <h5>Shop Name:</h5>
-              Shop Location: adress #3444 dpt 52
+              Shop Location: {user.address}
```
Comment (dvwhite) [inline src/components/pages/ProfilePage/GroomerProfilePage.js:42]: Great job removing the dummy data and hooking it up to the API data

**Review by dvwhite — APPROVED** [review]
Well done! 🤝

**Pull request merged**